src/sortStudents.ts

```
@@ -1,16 +1,54 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc'| 'desc';
 
+function averageGrade({ grades } : Student) :number {
+  return grades.reduce((sum, grade) => sum + grade, 0) / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortedStudents = [...students];
+
+  return sortedStudents.sort((firstStudent, secondStudent) : number => {
+    switch (sortBy) {
+      case (SortType.Name):
+      case (SortType.Surname):
+        return order === 'asc'
+          ? firstStudent[sortBy].localeCompare(secondStudent[sortBy])
+          : secondStudent[sortBy].localeCompare(firstStudent[sortBy]);
+
+      case (SortType.AverageGrade):
+        return order === 'asc'
+          ? averageGrade(firstStudent) - averageGrade(secondStudent)
+          : averageGrade(secondStudent) - averageGrade(firstStudent);
+
+      case SortType.Age:
+      case SortType.Married:
+        return order === 'asc'
+          ? Number(firstStudent[sortBy]) - Number(secondStudent[sortBy])
+          : Number(secondStudent[sortBy]) - Number(firstStudent[sortBy]);
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+      default:
+        return 0;
```

Default case should throw an error

---

Good point. Needs to be fixed.